zookeeper/core/component.py

```diff
@@ -173,14 +179,23 @@ def base_wrapped_fn(instance, name):
         if name in instance.__component_instantiated_field_values__:
             # Type-check, cache, and return the result.
             result = instance.__component_instantiated_field_values__[name]
-            _type_check_and_cache(instance, field, result)
+            _type_check_and_maybe_cache(instance, field, result)
+            if instance.__component_configured__:
+                del instance.__component_instantiated_field_values__[name]
```

Comment:
There's no need to keep two references to the same value, as after configuration we rely on the value being in the instance dict. I guess it doesn't matter too much since we don't allow delattr of field values, but if we did then keeping this reference would prevent values from being garbage collected.